### PR TITLE
Implement "paste" command for statement fragments

### PR DESCRIPTION
**Pull request by david-a-wheeler**
This is a start at implementing the "paste" command. I"ve stubbed out copying from the clipboard.

### CODE REVIEW

**Comment by david-a-wheeler** [comment]
I've been struggling to figure out how to get the string *from* the clipboard in ReScript, so I've stubbed it out.

It's not a lot of code, but my goal is to get better enough over time to be actually helpful.

Comments are of course quite welcome.

File: src/metamath/ui/MM_cmp_user_stmt.res [hunk 1]
```diff
@@ -1068,6 +1069,30 @@ let make = React.memoCustomCompareProps( ({
         }
     }
 
+    let actPasteFromClipboard = () => {
+        // TODO: Support undo/redo
```
Comment (expln) [inline src/metamath/ui/MM_cmp_user_stmt.res:1073]: Please don't leave TODOs in the final code. Undo for this action will be supported by the undo I am currently developing.

File: src/metamath/ui/MM_cmp_user_stmt.res [hunk 2]
```diff
@@ -1149,6 +1174,7 @@ let make = React.memoCustomCompareProps( ({
                     <MM_Icons.Logout style=ReactDOM.Style.make(~transform="rotate(90deg)", ()) />
                 </Button>
                 <Button title="Copy to the clipboard" onClick={_=>actCopyToClipboard()} ?style> <MM_Icons.ContentCopy/> </Button>
+                <Button title="Paste from the clipbboard to the selection" onClick={_=>actPasteFromClipboard()} ?style> <MM_Icons.ContentPaste/> </Button>
```
Comment (expln) [inline src/metamath/ui/MM_cmp_user_stmt.res:1177]: `clipbboard` -> `clipboard`

Comment (expln) [inline src/metamath/ui/MM_cmp_user_stmt.res:1177]: Please try not to exceed length of 120 characters for lines.  In some scenarios this is difficult to achieve, especially when lines are prepended with long empty space due to indentation. But here it is possible to write code like that:

```
<Button title="Paste from the clipboard to the selection" onClick={_=>actPasteFromClipboard()} ?style> 
    <MM_Icons.ContentPaste/> 
</Button>
```

**Review by expln — COMMENTED** [review]
Everything looks good. I left few small comments. And the below is how to change the code to read from the clipboard.

navigator.clipboard.readText() returns a promise https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/readText

So, the code should be changed like that:

MM_react_common.res
```
let readFromClipboard = ():promise<string> => {
    navigator["clipboard"]["readText"](.)
}
```

MM_react_common.resi
```
let readFromClipboard: () => promise<string>
```

MM_cmp_user_stmt.res
```
let actPasteFromClipboard = async () => {
    let clipboardContents = await readFromClipboard()
    let selectedRangeIndices = getSelectedRange()
    switch selectedRangeIndices {
        | None => ()
        | Some((low, high)) =>
            let currentText = stmt.cont->contToStr
            let newText = (
                currentText->Js.String2.slice(~from=0, ~to_=low) ++
                clipboardContents ++
                currentText->Js.String2.sliceToEnd(~from=high)
            )
            // Propagate changes to MM_cmp_editor
            onContEditDone(newText)
    }
}
```
```
onClick={_=>actPasteFromClipboard()->ignore}
```

File: src/metamath/ui/MM_cmp_user_stmt.res (hunk 1 above)
Comment (david-a-wheeler) [reply src/metamath/ui/MM_cmp_user_stmt.res:1073]: No problem. I just didn't want to ruin what you were doing :-).

**Comment by david-a-wheeler** [comment]
Thanks so much for the information & review. I *think* it's ready.

**Comment by expln** [comment]
Thank you David! I will deploy it to dev soon.

**Pull request merged**